crates/core/tests/helpers/deploy_helpers.rs

```diff
@@ -0,0 +1,476 @@
+        full_path
+    }};
+}
+
+#[derive(Debug, Deserialize)]
+struct Profile {
+    default: DefaultProfile,
+}
+
+#[derive(Debug, Deserialize)]
+struct DefaultProfile {
+    out: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct FoundryConfig {
+    profile: Profile,
+}
+
+fn get_foundry_default_out(file_path: &std::path::Path) -> Result<String, Box<dyn std::error::Error>> {
+    let mut file = File::open(file_path)?;
+    let mut contents = String::new();
+    file.read_to_string(&mut contents)?;
+
+    let config: FoundryConfig = toml::from_str(&contents)?;
+
+    // Since 'out' is a field in 'Default' struct, we can access it directly
+    Ok(config.profile.default.out)
+}
+
+// This assumes you are adding a solidity file in kakarot-rpc/solidity_contracts
+// and ran `forge build --names --force`
+pub fn get_contract(filename: &str) -> (Abi, ethers::types::Bytes) {
+    let dot_sol = format!("{filename}.sol");
+    let dot_json = format!("{filename}.json");
+    let foundry_toml_path = root_project_path!("foundry.toml");
+    let foundry_default_out = get_foundry_default_out(&foundry_toml_path).unwrap();
+    let compiled_solidity_path = std::path::Path::new(&foundry_default_out).join(dot_sol).join(dot_json);
+    let compiled_solidity_path_from_root = root_project_path!(&compiled_solidity_path);
+
+    // Read the content of the file
+    let contents = fs::read_to_string(compiled_solidity_path_from_root).expect("{filename} isn't compiled.");
+
+    // Parse the entire JSON content into a Value
+    let json: serde_json::Value = serde_json::from_str(&contents).unwrap();
+
+    // Extract the `abi` field and parse it into the Abi type
+    let abi: Abi = serde_json::from_value(json["abi"].clone()).unwrap();
+
+    // Extract the `bytecode` field
+    let bytecode_str = json["bytecode"]["object"].as_str().unwrap();
+    let bytecode_vec = hex::decode(bytecode_str.trim_start_matches("0x")).unwrap();
+    let bytecode = ethers::types::Bytes::from(bytecode_vec);
+
+    (abi, bytecode)
+}
+
+// extracted/simplified logic from ethers https://github.com/gakonst/ethers-rs/blob/master/ethers-contract/src/factory.rs#L385
+// otherwise we would have to mock an eth client in order to access the logic
+pub fn encode_contract<T: Tokenize>(
+    abi: &Abi,
+    bytecode: &ethers::types::Bytes,
+    constructor_args: T,
+) -> ethers::types::Bytes {
+    let params = constructor_args.into_tokens();
+    match (abi.constructor(), params.is_empty()) {
+        (None, false) => {
+            panic!("Contract does not take arguments in its constructor, but passed user params are not empty.")
+        }
+        (None, true) => bytecode.clone(),
+        (Some(constructor), _) => {
+            let res = constructor.encode_input(bytecode.to_vec(), &params);
+            res.unwrap().into()
+        }
+    }
+}
+
+// Constructs an ethereum transaction with the correct Kakarot chain ID, and default values
+// for everything but nonce, to, and bytes
```

Comment:
why is `to` field not an Address?

Reply:
Ahh I think I understand, because TransactionKind::create will probably be None or smth

Reply:
yeah, reth implements this as an enum type: https://github.com/paradigmxyz/reth/blob/main/crates/primitives/src/transaction/mod.rs#L716